pkg/addon/addon.go

```diff
@@ -215,8 +215,8 @@ func ListAddonUIDataFromReader(r AsyncReader, registryMeta map[string]SourceMeta
 			}
 			addonRes.RegistryName = registryName
 			l.Lock()
+			defer l.Unlock()
 			addons = append(addons, addonRes)
-			l.Unlock()
```

Comment:
the old way is more clear to me, why must we use defer here? It's more like to cause a bug as there're outside for-loop.